libr/include/r_util/r_annotated_code.h

```diff
@@ -79,105 +79,92 @@ typedef struct r_annotated_code_t {
+ * @return Pointer to the new RAnnotatedCode structure created.
  */
 R_API RAnnotatedCode *r_annotated_code_new(char *code);
 /**
- * r_annotated_code_free() - Deallocates *code.
- * @code: Pointer to a RAnnotatedCode.
- *
  * This functions deallocates memory allocated for *code.
+ * @brief Deallocates *code.
  * 
- * Return: Nothing.
+ * @param code Pointer to a RAnnotatedCode.
  */
 R_API void r_annotated_code_free(RAnnotatedCode *code);
 /**
- * r_annotation_free() - Deallocates dynamically allocated memory for the specified annotation.
- * @e: Pointer to the annotation
- * @user: Always NULL for this function. Present here for this function to be of the type RVectorFree.
- * 
  * This function recognizes the type of the specified annotation and
  * frees memory that is dynamically allocated for it.
+ * @brief Deallocates dynamically allocated memory for the specified annotation.
  * 
- * Return: Nothing.
+ * @param e Pointer to the annotation.
+ * @param user Always NULL for this function. Present here for this function to be of the type RVectorFree.
  */
 R_API void r_annotation_free(void *e, void *user);
 /**
- * r_annotation_is_reference() - Checks if the specified annotation is a reference.
- * @annotation: Pointer to an annotation.
- * 
  * This function recognizes the type of the specified annotation and returns true if its
  * type is any of the following three: R_CODE_ANNOTATION_TYPE_GLOBAL_VARIABLE,
  * R_CODE_ANNOTATION_TYPE_CONSTANT_VARIABLE, R_CODE_ANNOTATION_TYPE_FUNCTION_NAME
+ * @brief Checks if the specified annotation is a reference.
  * 
- * Return: Returns true if the specified annotation is a reference.
+ * @param annotation Pointer to an annotation.
+ * @return Returns true if the specified annotation is a reference.
  */
 R_API bool r_annotation_is_reference(RCodeAnnotation *annotation);
 /**
- * r_annotation_is_variable() - Checks if the specified annotation is a function variable.
- * @annotation: Pointer to an annotation.
- * 
  * This function recognizes the type of the specified annotation and returns true if its
  * type is any of the following two: R_CODE_ANNOTATION_TYPE_LOCAL_VARIABLE,
  * R_CODE_ANNOTATION_TYPE_FUNCTION_PARAMETER
+ * @brief Checks if the specified annotation is a function variable.
  * 
- * Return: Returns true if the specified annotation is a function variable.
+ * @param annotation Pointer to an annotation.
+ * @return Returns true if the specified annotation is a function variable.
  */
 R_API bool r_annotation_is_variable(RCodeAnnotation *annotation);
 /**
- * r_annotated_code_add_annotation() - Inserts *annotation in *code.
- * @code: Pointer to a RAnnotatedCode.
- * @annotation: Pointer to a annotation.
- *
  * This functions inserts the annotation represented by the pointer 'annotation' to the vector
  * of annotations in the RAnnotatedCode represented by 'code'. To be more precise,
  * annotation will be added to code->annotations, which is a RVector<RCodeAnnotation> annotations.
```

Comment:
There is no need to go into implementation details like "added to code->annotations" which is RVector<RCodeAnnotation>" unless it affects the way this function used. The whole point of this function is to hide implementation detail like this so that it in future it can be easily replaced with different container or multiple of them if necessary to answer queries about annotated code. For example it might replaced with a specialized tree structure which allows implementing `r_annotated_code_annotations_in` more efficiently.

Reply:
Thanks, I have updated this.